test/spoof/error_checks_test.go

```diff
@@ -57,31 +58,53 @@ func TestDNSError(t *testing.T) {
 	}
 }
 
-func TestTCPConnectRefuse(t *testing.T) {
+func TestConnectionRefused(t *testing.T) {
 	client := &http.Client{}
 
 	for _, tt := range []struct {
-		name      string
-		url       string
-		tcpRefuse bool
+		name        string
+		url         string
+		connRefused bool
 	}{{
 		name:      "nothing listening",
 		url:       "http://localhost:60001",
-		tcpRefuse: true,
+		connRefused: true,
 	}, {
 		name:      "dns error",
 		url:       "http://this.url.does.not.exist",
-		tcpRefuse: false,
+		connRefused: false,
 	}, {
 		name:      "google.com",
 		url:       "https://google.com",
-		tcpRefuse: false,
+		connRefused: false,
 	}} {
 		t.Run(tt.name, func(t *testing.T) {
 			req, _ := http.NewRequest("GET", tt.url, nil)
 			_, err := client.Do(req)
-			if tcpRefuse := isTCPConnectRefuse(err); tt.tcpRefuse != tcpRefuse {
-				t.Errorf("Expected tcpRefuse=%v, got %v", tt.tcpRefuse, tcpRefuse)
+			if connRefused := isConnectionRefused(err); tt.connRefused != connRefused {
+				t.Errorf("Expected connRefused=%v, got %v", tt.connRefused, connRefused)
+			}
+		})
+	}
+}
+
+func TestConnectionReset(t *testing.T) {
```

Comment:
Like this unit test better than the one above, IMHO the one above should also be something like this test

Reply:
Well, on one hand, the one above is doing more than just checking errors (using HttpClient) but at the same time it uses the real errors from Go library.